cpp/src/interop/detail/arrow_allocator.cpp

```diff
@@ -30,7 +61,7 @@ std::unique_ptr<arrow::Buffer> allocate_arrow_buffer(int64_t const size, arrow::
   */
   auto result = arrow::AllocateBuffer(size, ar_mr);
   CUDF_EXPECTS(result.ok(), "Failed to allocate Arrow buffer");
-  return std::move(result).ValueOrDie();
+  return enable_hugepage(std::move(result).ValueOrDie());
```

Comment:
What is the reason for `std::move` here? I'm not sure you need that.

Generally `return std::move(...)` is an antipattern. When you are returning a local variable, usually the copy will be elided anyway. If there is a move ctor, it can be used automatically.

Reply:
Notice, we use `std::move` on the intermediate `result`, not the value returned by `ValueOrDie()`. This _moves_ the value (`std::unique_ptr<arrow::Buffer>`) out of `result` .
I have added explicit types to make this more clear.

Reply:
Right, I was assuming you just copied the std::move from the code that was there before, which just had `return std::move`, which I think was unnecessary (and disables the compiler from eliding the move with RVO, IIUC).

Reply:
The compiler does not allow `return std::move(retval)` in libcudf because it disables RVO. But `std::move(result).ValueOrDie()` is slightly different, as it does not move the return value, just casts `result` to rvalue, presumably so that `ValueOrDie` can move the value out of `result`. The moved value is most likely then RVOd to the caller (not sure how this is actually called :)).